### DISCUSSION TITLE
License ?

### DISCUSSION THREAD

**Discussion by suprafun** [visose/Robots#79] [Q&A] [answered]
Hello, under what license is this project released under ? I would like to study it so I can learn how to use the Rhino API. Thank-you.

**Comment by visose** [comment, marked as answer]
I would release the original code under MIT license, unless any of the projects I borrowed code from requires this whole project to adopt a more restrictive license. You can find a list of them in the about section. Mainly the IK code for UR robots is from ROS, which is released under a Creative Commons 3.0 license. Still very permissive.

If you're asking regarding documentation for the API, I'm afraid there is none, there's only info about the Grasshopper components in the wiki. In any case, the project is mainly a .NET library with a Grasshopper wrapper. I use it as a programming library more often than the actual Grasshopper components. You just need to reference Robots.dll to your project. I can post some example code on how to consume it if you're interested.
